### PR TITLE
src: DragonFlyBSD has mgshdr struct too

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by santigimeno — APPROVED** [review]

**Review by cjihrig — APPROVED** [review]

**Comment by bnoordhuis** [comment]
Dragonfly doesn't have `sendmmsg()` and `recvmmsg()` though, does it? It didn't last time I researched it and a quick google search doesn't turn up anything now either.

**Comment by devnexen** [comment]
sure but it was more to fix the build (I spotted the warnings) not to support these functions I did not change the ifdef conditions anyway.

**Comment by bnoordhuis** [comment]
What's the warning? Turning on `HAVE_MMSG` means libuv will try to use `sendmmsg()` and `recvmmsg()` - note the double 'm' - but that doesn't make sense for dragonfly.

The system calls are currently gated on` __FreeBSD__ >= 11` so that's probably why you're not seeing actual build errors.

**Comment by devnexen** [comment]
> 
> The system calls are currently gated on` __FreeBSD__ >= 11` so that's probably why you're not seeing actual build errors.

Exactly what I ve said I did not change these. The build complains the uv__mmsghdr not defined which DragonFlyBSD but the libuv wrappers still return ENOSYS as not implemented (yet).

**Comment by bnoordhuis** [comment]
Then I suggest rewriting your patch because this approach will break when support for freebsd < 11 is dropped. There are no dragonfly machines in the CI matrix, breakage will go unnoticed.

File: src/unix/freebsd.c [hunk 1]
```diff
@@ -265,7 +265,7 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
 
 
 int uv__sendmmsg(int fd, struct uv__mmsghdr* mmsg, unsigned int vlen) {
-#if __FreeBSD__ >= 11
+#if __FreeBSD__ >= 11 && !__DragonFly__
```
Comment (bnoordhuis) [inline src/unix/freebsd.c:268]:
```suggestion
#if __FreeBSD__ >= 11 && !defined(__DragonFly__)
```

File: src/unix/freebsd.c [hunk 2]
```diff
@@ -274,7 +274,7 @@ int uv__sendmmsg(int fd, struct uv__mmsghdr* mmsg, unsigned int vlen) {
 
 
 int uv__recvmmsg(int fd, struct uv__mmsghdr* mmsg, unsigned int vlen) {
-#if __FreeBSD__ >= 11
+#if __FreeBSD__ >= 11 && !__DragonFly__
```
Comment (bnoordhuis) [inline src/unix/freebsd.c:277]:
```suggestion
#if __FreeBSD__ >= 11 && !defined(__DragonFly__)
```
(otherwise it'll emit preprocessor warnings on freebsd)

**Review by bnoordhuis — APPROVED** [review]
LGTM. I can probably fix/commit the nits myself.

**Comment by bnoordhuis** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/2087/

File: src/unix/internal.h [hunk 3]
```diff
@@ -327,7 +327,8 @@ int uv__getsockpeername(const uv_handle_t* handle,
 
 #if defined(__linux__)            ||                                      \
     defined(__FreeBSD__)          ||                                      \
-    defined(__FreeBSD_kernel__)
+    defined(__FreeBSD_kernel__)   ||					  \
```
Comment (vtjnash) [inline src/unix/internal.h:330]:
```suggestion
    defined(__FreeBSD_kernel__)   ||                                       \
```
(tabs-to-spaces)

**Comment by vtjnash** [comment]
https://ci.nodejs.org/job/libuv-test-commit/2150/

**Pull request merged**